rescan.go

```diff
@@ -168,7 +206,7 @@ func updateChan(update <-chan *updateOptions) RescanOption {
 
 // rescan is a single-threaded function that uses headers from the database and
 // functional options as arguments.
-func (s *ChainService) rescan(options ...RescanOption) error {
+func (r *Rescan) rescan(options ...RescanOption) error {
```

Comment:
could also be made functional as `func rescan(chain ChainSource, options ...RescanOption) error` ?